### DISCUSSION TITLE
Showcase for games made with Flare?

### DISCUSSION THREAD

**Discussion by WithinAmnesia** [flareteam/flare-game#913] [Show and tell]
What games have been developed / worked on with the Flare Engine / Flare Assets?

For example I have been prototyping / tinkering away with a large scale A.R.P.G. with the Flare Engine and Flare Assets:
![Working Ramps Success!!!](https://user-images.githubusercontent.com/86871191/178803482-1e12a52a-ca43-46fb-907d-91fea246077c.png)
![Treasure Trunks Working!!!](https://user-images.githubusercontent.com/86871191/178803841-2076d565-50a8-45ff-9be1-1a10aa1a1721.png)
![Michael Kh'thulis (Blacksmith)!](https://user-images.githubusercontent.com/86871191/178803873-fe16eea3-aa31-4ce4-8431-7a71c066b295.png)
![Spell Sword Works!](https://user-images.githubusercontent.com/86871191/178804040-4c90d528-27f0-41a9-8a68-8b36ff29ae04.png)
![Dual-Wielding Works Great!](https://user-images.githubusercontent.com/86871191/178804114-5e670959-c8d1-4afa-b7ca-059947066263.png)
![Wonderous Waterfall Work-In-Progress!](https://user-images.githubusercontent.com/86871191/178804171-a030382f-8699-45be-b5b1-b9b91fefc04e.png)
![Justin the Giant! -'I  Justin  was able to get all 5 to load (proof)  This was with a 64-bit Linux build '](https://user-images.githubusercontent.com/86871191/178804464-17e1e1a3-8540-4c11-bd6b-5b319a3ccb37.png)
![4K 3840×2160-16;9 Resolution!](https://user-images.githubusercontent.com/86871191/178804486-bc9dd7e5-0e00-4768-9670-080cffe9c0a5.png)
![Stress Test 1,709-6,836](https://user-images.githubusercontent.com/86871191/178804546-31db553e-b0d6-48a1-985e-98c0e47161d4.png)
https://www.deviantart.com/withinamnesia/art/The-Jewelled-Eye-of-AGABAL-Jeweyel-915001367

What ~Flare games (released or in development) are out there that the community knows of? What are some examples of games developed / being developed with Flare?

**Comment by dorkster** [comment]
As a baseline, I try to keep a comprehensive list of everything that is using / has used Flare on the [Mods](https://flarerpg.org/mods/) page. In-development and abandoned mods are listed on the [Other Mods](https://flarerpg.org/mods/other-mods/) sub-page.

It'd be interesting to see if there were any others. If anyone has a project they're working on, feel free to share a Github/Gitlab/forum/website link, and I'll add it to the list.

**Reply by WithinAmnesia** [reply]
What about that / those (asian? / Chinese?) mobile game(s?) that used Flare assets [I forgot the name(s)]? Does that count?

**Reply by dorkster** [reply]
The only one I know of is FlareX/Kingdom Quest. I don't promote it because:

- They never released their source changes in any capacity, and are thus violating the GPL. They have since added links back to our Github, but that is not sufficient when changes to the source code are made.
- There's ads during loading screens.
  - There's also Google Play integration, which I think many in the libre game community would consider an "anti-feature".
- The game data itself is just our Empyrean Campaign with a few minor changes.
- The engine and game data are out of date. Makes it annoying when I get bug reports for things that were fixed several versions ago in the official engine/game.

**Reply by WithinAmnesia** [reply]
Okay so they are not keeping with the good will of the community / are doing some shady business practices I take it.

I wonder if anyone else has spotted more Flare assets ~'in the wild'? I hope to make a first commercially viable game kind of like a big Diablo 1 with some overworld (Fields, Castle, Towns, Caves, Cemetery, Woods etc.). Although I hope to keep it all above board and keeping with the good will of the community and have everything be working in good faith.

Also I hope to have a one third free / shareware demo / old school style free demo where it is basically the first third of the game for free to try out and see if anyone likes it enough to buy the full thing; like Doom back in the day. If the thing gets a release and some revenue I hope to get more assets made and release them open source. So I can kind of come full circle from a modder getting cease and desist (working on Blizzard games especially) to hopefully basically doing the opposite and making it easier / empowering creative people to go from modder to professional game developer (if they make good enough stuff for people to enjoy on the open market).

I am applying for registered trade mark so my decades of work has a (hopefully) strong legal fortification and I have a fair chance at getting paid for working 4-6 years directly on a game (more like 2+ decades indirectly) / not have my stuff be stolen and then sold 'as is' / a carbon copy of my exact game on some random online game store. I don't even mind if people straight up pirate the game if they are broke etc. but I am not going to take straight up thieves stealing my work and selling it; that is too much. I can make free open source assets, features and all the tools and things people needed to make their games and dreams become a reality until the cows come home but no one should take abuse and have their work be illegally stolen and resold in bad faith and dishonesty; that just seems wrong and not aiming after something noble.

That being said I hope that there can be games made with Flare that are commercially viable big or small that hopefully reward honest effort and game design and that the wider global market / gaming community can enjoy. I hope people @dorkster , @Danimal696 Clint and others here slogging away year after year could see more success / nurture a community with really good games that are honest and straight shooting (unlike a certain recent Blizzard mobile A.R.P.G.). I think more than ever today there is a desire / need for regular people to be able to make high quality games that they want to play without needing a 300+ corporate team with endless crunches and deadlines. I kind of hope Flare can become something like a light in the darkness and empower people to make really special / high quality games from just individuals dreaming to make amazing games worth playing. As in kind of like realizing personal sovereignty or being the master of one's own destiny; being forever free.

It feels almost like: To be strong for others, one must first be strong themselves.

**Comment by WithinAmnesia** [comment]
Where are / what happened to @Smujb , @Danimal696 , @makrohn , [Rixty](https://opengameart.org/users/rixty) (I can't find their GitHub profile link) and @clintbellanger (I think Clint is busy with work now; I hope Wandercall can be further developed). I think @dorkster has (a) project(s) of some sort but my brain is showing myself ~Starcraft / Dune I / Dune II RTS mixed with Space Quest with Command and Conquer talking heads on a zoom call and I am not very confident that has anything to do with Justin's project(s) XD).

I wonder how many projects are being worked on and off currently with Flare and what the aspirations / hopes / dreams of the creators are for them.

I think each of the above creators tried new things / features / art assets and made more discoveries as to what Flare could do. I wonder what the current state / future plans are for the creative community with Flare.

**Reply by Danimal696** [reply]
So far I´m awaiting for the new features of random loot and lv scalable enemies; don´t plan to touch Flare again until those are out. I still drop by from time to see if any progress was made.

**Reply by dorkster** [reply]
@Danimal696 FYI, https://github.com/flareteam/flare-engine/issues/1784 has been sort of a blocker in terms of the roadmap of planned features (particularly for 1.14). I've been quietly working on it in the background, and progress is good. I think I'll be able to put it to rest sooner rather than later.

I think the only remaining task for level scaled enemies is a way to scale the loot rewards. So that means https://github.com/flareteam/flare-engine/issues/1816 and (at least partially) https://github.com/flareteam/flare-engine/issues/1815 will need implementing. Not sure at the moment about committing to these for 1.14.

**Reply by WithinAmnesia** [reply]
I am in the same boat more or less. I'm hoping to get the new itemization / stat mechanics hammered out to form a core for my project as in the ~'beating heart' of main gameplay progression 'loop' / 'player progression mechanisms'. Those more or less being player levels, stats, items, abilities, quests with a full level range of enemies to fight with level scaling experience rewards.

Right now I am trying to get the core gameplay elements in a state that can be used as a foundation for future builds for the future. With the aim for gameplay progression elements  as in player level and item progression to be immediately worked on after the new stats and itemization are ready and properly working. I hope to first focus on and prioritize the core gameplay and progression elements (as in stats and itemization, abilities, enemies, quests) that make a game experience ~'deep'. I can always making my game more ~'wide' with more levels and graphics but I only really have a few proper shots at making my game ~'deep'. Thus I am prioritizing the 'deep' core features over the expandable 'wide' features.

Then once after I have the ~'deep' core to my who gameplay experience hammered out properly I then wish to focus on going ~'wide'. As in to work ~'wide' as in working on extrapolating / painting / mass producing the rough prototype area graphics enough to support a pre-alpha for a 'blocked out' testing environment that uses quickly made graphics but built of elements that can act as a foundation for a future builds and later quality polish down the stretch. I see no sense in using the current game engine stat and itemization mechanics only to be forced into redoing the core heart of the game. In fact I was forced to do this with my last killed off RPG Maker MV project for I needed to redo 'temporary' stat and itemization mechanics that turned out to be a complete nightmare where it was better to just leave 3-5 years of work behind and start on a different platform all together.

That level of disaster should not happen here but I had to learn the hard way that get the priorities together straight and work on the core foundational features first directly in a line and only experiment with graphics and other elements in small scale as proof of concept for future ways of using the development tools available to one's self. Instead of making the art, cosmetics and decorations amazing then realizing your castle foundation was built upon sand and the tide is coming in.

Still though good things take time. I would rather wait a year for a proper level of quality of feature used as the foundation of the entire game moving forward than to rush something out that lack quality and must be 'put up with' for years and years. Do it right the first time and one will not have to keep redoing it over and over.

Quality really matters for foundations be it in with games or even building a house or a castle. All of your work depends on the strength of the foundations. So if anything should be the first priority of high quality work it should be the foundations; the followed by the roof and water proofing, wiring and plumbing and structural supporting elements (frame) and then once the skeleton and 'flesh and blood' are on then one can worry about the cosmetics and the 'clothing'. For Flare stats and itemization are pretty much the beating heart of the whole game genre. The better heart we get the stronger our games can operate and hopefully the fitter and more powerful our gameplay experiences can be experienced for our future players.

**Reply by WithinAmnesia** [reply]
@dorkster Would it make sense for future ~'1.15' to focus on item generation like with https://github.com/flareteam/flare-engine/issues/1815 + https://github.com/flareteam/flare-engine/issues/1816 and maybe possibly something with https://github.com/flareteam/flare-engine/issues/1817 (I think more than myself would like ~'sockets' / retroactive item adjustments / upgrades)?

Also I think it makes sense to meticulously progress with 1.14 (with it being done only when it tells when it is done; it takes the time that it takes).

Are there any plans for ~'1.15' that I am not thinking of off the top of my head?

**Reply by dorkster** [reply]
>Would it make sense for future ~'1.15' to focus on item generation

In short, I'd say yes. I'm on board with random item generation. Less sure about socketed items.
As for now, I don't have anything else significant planned for 1.15.

**Reply by WithinAmnesia** [reply]
Socketed / retroactively upgrade-able items are cool but like, complexity. It does not seem like an easy thing to do. I suppose if it was pulled off *somehow* that it could also work as a basis / foundation for 'enchanting' or 'blessing' or 'cursing' or 'augmenting' items.

As in like in other RPGs where say using a ~'magical stone' on a weapon gives it +1% damage boost (or something). Or adding a 'magical patch' to piece of armour gives it +1% armour boost (or something). As in to be able to mix and match / ~'customize' equipment as a sort of second tier of itemization.

It works kind of like where appropriate equipment slots have sort of ~'items on top of items' that fuel the item hunt. As in like say a side quest gives an item augmentation (like a priest giving ~'blessed cord' (or something)) it could make a game have more depth in more areas and give more relevance to more levels and side aspects of the game. Kind of of like back tracking but not forced but as extra options for players to explore and experiment with in a natural / intuitive fashion.

I remember in vanilla WoW the major raids had the best items to equip. Yet there were a few minor raids / dungeons that offered item augmentations that were best in slot. So both major and minor pieces of content were made relevant and offered a fun excuse to go back to areas where naturally a player would not go back to once power-creep took affect. I kind of remember also Metroid and Castlevania style RPGs had areas in early / lower level areas that circled back to enhanced late game items or elements or progression. I always thought that was kind of neat / good game design.

Or like in Diablo II where people farm the Countess in a typically (more or less) set of lower level areas to farm up middle level to some High Runes to fuel item augmentations to later progress better in higher level areas. Or Lower Kurast runs for chests in hopes to getting High Runes to also give late game that extra push via item augmentations. Here are some videos of what a lot of Diablo II players do to farm item augmentations / Middle to High Runes. https://www.youtube.com/watch?v=rqeAVY8DcXE&ab_channel=Dbrunski125 + https://www.youtube.com/watch?v=3hldedCpaO0&ab_channel=Dbrunski125 + https://diablo.fandom.com/wiki/Gems + https://diablo.fandom.com/wiki/Runes + https://diablo.fandom.com/wiki/Category:Socketable_Items + https://diablo.fandom.com/wiki/Horadric_Cube . Also Path of Exile is all about those item augmentations as a core gameplay driver: https://pathofexile.fandom.com/wiki/Guide:Crafting + https://pathofexile.fandom.com/wiki/Labyrinth_enchantment + https://pathofexile.fandom.com/wiki/Crafting_Bench + https://pathofexile.fandom.com/wiki/Item + https://pathofexile.fandom.com/wiki/Chaos_Orb + https://pathofexile.fandom.com/wiki/Divine_Orb + https://pathofexile.fandom.com/wiki/Vaal_Orb + https://pathofexile.fandom.com/wiki/Orb_of_Alteration + https://pathofexile.fandom.com/wiki/Orb_of_Transmutation + https://pathofexile.fandom.com/wiki/Regal_Orb + https://pathofexile.fandom.com/wiki/Currency + WoW Item Enchanting: https://classic.wowhead.com/enchanting

**Reply by Danimal696** [reply]
> @Danimal696 FYI, [flareteam/flare-engine#1784](https://github.com/flareteam/flare-engine/issues/1784) has been sort of a blocker in terms of the roadmap of planned features (particularly for 1.14). I've been quietly working on it in the background, and progress is good. I think I'll be able to put it to rest sooner rather than later.
> 
> I think the only remaining task for level scaled enemies is a way to scale the loot rewards. So that means [flareteam/flare-engine#1816](https://github.com/flareteam/flare-engine/issues/1816) and (at least partially) [flareteam/flare-engine#1815](https://github.com/flareteam/flare-engine/issues/1815) will need implementing. Not sure at the moment about committing to these for 1.14.


It´s good to know you are still working on it; as for the scaling, just passing the lv of the monster to the item generator should suffice.

**Reply by WithinAmnesia** [reply]
I remember in vanilla WoW end game 5-man dungeon bosses tended to drop items ~3-5 item levels / required levels behind their own (Say level 63 boss drops a rare item that requires Level 58 (even though the max level was 60)).

Could a arbitrary level value / range of levels given / passed to the item generation work? So it kind of works like who @Danimal696 was saying but also able for developers to tweak / adjust the range / item level(s) given to the random item generator? Or would like random drop tables work for this anyway and be sort of simplification / ~'less moving parts'?

Then again it might not be so simple if ~100 different levels and ranges are involved for loot drop tables. Hmm what are your thoughts on this (I'm kind of pooped after today working on the mountain so there might be some missing points)? I like that kind of option though being able to send (a range of) monster level(s) to a random item generator / item loot table.
